volatility3/framework/plugins/linux/lsblk.py

```diff
@@ -0,0 +1,388 @@
+            return 0
+        while hlist_node:
+            inode = linux.LinuxUtilities.container_of(hlist_node, "inode", "i_hash", vmlinux)
+            # These checks are from the find_inode function https://elixir.bootlin.com/linux/v5.4/source/fs/inode.c#L805
+            # Trying to find the inode with the correct device number and calculated inode number from the inode cache
+            if inode.i_sb == blockdev_superblock and inode.i_rdev == devt:
+                break
+            hlist_node = hlist_node.next
+
+        return inode.i_bdev
+
+    def _device_iterator(self, context, klist_devices):
+        vmlinux = context.modules[self.config["kernel"]]
+        # Linux v5.1 moves device->knode_class to device_private->knode_class
+        knode_class_in_private_device = vmlinux.get_type("device_private").has_member(
+            "knode_class"
+        )
+
+        if knode_class_in_private_device:
+            for klist_node in klist_devices.k_list.to_list(
+                vmlinux.symbol_table_name + constants.BANG + "klist_node", "n_node"
+            ):
+                device_private = linux.LinuxUtilities.container_of(klist_node.vol.offset, "device_private", "knode_class", vmlinux)
+                yield device_private.device.dereference()
+        else:
+            for klist_node in klist_devices.k_list.to_list(
+                vmlinux.symbol_table_name + constants.BANG + "klist_node", "n_node"
+            ):
+                device = linux.LinuxUtilities.container_of(klist_node.vol.offset, "device", "knode_class", vmlinux)
+                yield device
+
+    def _get_name(self, vmlinux, block_device, gendisk):
+        # block_device does not have bd_partno before 5.11
+        if vmlinux.get_type("block_device").has_member("bd_partno"):
+            if block_device.bd_partno:
+                return utility.array_to_string(gendisk.disk_name) + str(
+                    block_device.bd_partno
+                )
+        else:
+            if gendisk.part0.partno:
+                return utility.array_to_string(gendisk.disk_name) + str(
+                    gendisk.part0.partno
+                )
+
+        if utility.array_to_string(gendisk.disk_name).startswith("dm-"):
+            if vmlinux.has_type("mapped_device"):
+                mapped_device = vmlinux.object(
+                    object_type="mapped_device",
+                    offset=gendisk.private_data,
+                    absolute=True,
+                )
+
+                hash_cell = vmlinux.object(
+                    object_type="hash_cell",
+                    offset=mapped_device.interface_ptr,
+                    absolute=True,
+                )
+
+                if hash_cell.name:
+                    return utility.pointer_to_string(hash_cell.name, 32)
+
+        return utility.array_to_string(gendisk.disk_name)
+
+    def _get_type(self, context, block_device, gendisk):
+        vmlinux = context.modules[self.config["kernel"]]
```

Comment:
Rather than limiting this function to being usable during a run of this plugin, could we add a parameter for `kernel_module_name` please?  This would then allow a lot of these to become classmethods, and make then usable by other plugins in the future.